src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1010,8 +1010,19 @@ private List<AssociativeNode> GetModifiedNodes(Subtree subtree, bool redefinitio
                     // It can then be handled normally regardless of its ForceExecution state
                     subtree.ForceExecution = false;
 
+                    BinaryExpressionNode prevBNE = null;
+                    if (node is BinaryExpressionNode bne1){
+                        var prevNode = st.AstNodes.OfType<BinaryExpressionNode>().Where(x => (x.LeftNode as IdentifierNode).Name == bne1.LeftNode.Name);
+                        if(prevNode.Count() > 1)
+                        {
+                            throw new Exception("1 prevnode expected");
+                        }
+                        prevBNE = prevNode.FirstOrDefault();
+                    }
+
                     //Check if the subtree (ie a node in graph) is an input and has primitive Right hand Node type
-                    if (redefinitionAllowed && st.IsInput && node is BinaryExpressionNode bne  && CoreUtils.IsPrimitiveASTNode(bne.RightNode))
+                    if (redefinitionAllowed && st.IsInput && node is BinaryExpressionNode bne  && CoreUtils.IsPrimitiveASTNode(bne.RightNode)
```

Comment:
If we don't want to make null node a primitive, we can remove the condition of bne.RightNode being a primitive I think.

Reply:
I also think we can remove the `node is BinaryExpressionNode bne` condition as it is implied if `CoreUtils.IsPrimitiveASTNode(prevBNE.RightNode` is true